server/src/main/java/com/cloud/network/element/ConfigDriveNetworkElement.java

```diff
@@ -630,6 +630,10 @@ private DataStore getDatastoreForConfigDriveIso(DiskTO disk, VirtualMachineProfi
     private boolean deleteConfigDriveIso(final VirtualMachine vm) throws ResourceUnavailableException {
         Long hostId  = (vm.getHostId() != null) ? vm.getHostId() : vm.getLastHostId();
         Location location = getConfigDriveLocation(vm.getId());
```

Comment:
@Pearl1594 
move `Location location = getConfigDriveLocation(vm.getId());` to line 637 (just before `if (location == Location.HOST) {`) ?